tap-agent/src/agent/sender_fee_tracker.rs

```diff
@@ -92,8 +109,8 @@ impl SenderFeeTracker {
     pub fn update(&mut self, id: Address, fee: u128) {
         if fee > 0 {
             // insert or update, if update remove old fee from total
-            if let Some(old_fee) = self.id_to_fee.insert(id, fee) {
-                self.total_fee -= old_fee;
+            if let Some(old_fee) = self.id_to_fee.insert(id, FeeCounter { fee, count: 0 }) {
+                self.total_fee -= old_fee.fee;
```

Comment:
It needs to have an update here for `pub fn update(&mut self, id: Address, fee: u128, counter: usize)`. SQL query and UnaggregatedReceipts need to be updated as well.